### PR TITLE
Improve `groupings` and `network` coverage

### PR DESCRIPTION
**Pull request by gnn**
The test coverage of these modules is a bit on the low side, so this branch collects some additional tests improving it.

### CODE REVIEW

File: oemof/groupings.py [hunk 1]
```diff
@@ -113,26 +115,26 @@ def __init__(self, key=None, constant_key=None, filter=None, **kwargs):
...
 
         Return :obj:`None` if you don't want to store :obj:`e` in a group.
         """
-        raise NotImplementedError(
-            "There is no default implementation for `Groupings.key`.\n" +
-            "Congratulations, you managed to execute supposedly " +
-            "unreachable code.\n" +
-            "Please let us know by filing a bug at:\n\n    " +
+        raise NotImplementedError("\n\n"
+            "There is no default implementation for `Groupings.key`.\n"
```
Comment (p-snft) [inline oemof/groupings.py]: I am not sure whether explicit concatenation was better.

File: oemof/network.py [hunk 2]
```diff
@@ -257,12 +259,12 @@ def __init__(self, input=None, output=None, flow=None, values=None,
             **kwargs):
         if flow is not None and values is not None:
             raise ValueError(
-                    "`Edge`'s `flow` and `values` keyword arguments are " +
-                    "aliases of each other, so they're mutually exclusive.\n" +
-                    "You supplied:\n"+
-                    "    `flow`  : {}".format(flow) +
-                    "    `values`: {}".format(values) +
-                    "\nChoose one.")
+                    "\n\n`Edge`'s `flow` and `values` keyword arguments are "
```
Comment (p-snft) [inline oemof/network.py]: Above, you delete double newlines, here you add some.

File: oemof/groupings.py [hunk 3]
```diff
@@ -9,15 +9,17 @@
...
-    from collections.abc import (Hashable, Iterable, Mapping,
-                                 MutableMapping as MuMa)
-except ImportError:
-    from collections import (Hashable, Iterable, Mapping,
+from collections.abc import (Hashable, Iterable, Mapping,
                              MutableMapping as MuMa)
 from itertools import chain, filterfalse
 
 
+# TODO: Update docstrings.
```
Comment (p-snft) [inline oemof/groupings.py]: Is this done? They seem to use nodes, now.

File: oemof/groupings.py (hunk 1 above)
Comment (p-snft) [reply oemof/groupings.py]: I have seen this is consistent with other parts of the code.

File: oemof/network.py (hunk 2 above)
Comment (p-snft) [reply oemof/network.py]: Seems to be consistent with other parts of the code.

File: oemof/groupings.py (hunk 3 above)
Comment (gnn) [reply oemof/groupings.py]: Eight lines later, `Entity` is still used. So that `TODO` can unfortunately not be ticked off yet.

**Comment by gnn** [comment]
This can be merged to get the new tests into v0.3.0.
I'll continue to improve coverage, but it's nice to already get some of these tests into the release cycle.

**Pull request merged**